src/layoutrestorer/layoutrestorer.ts

```diff
@@ -50,16 +66,69 @@ interface ILayoutRestorer {
   /**
    * Add a widget to be tracked by the layout restorer.
    */
-  add(widget: Widget, name: string): void;
+  add(widget: Widget, name: string, options?: ILayoutRestorer.IAddOptions): void;
 
   /**
-   * Wait for the given promise to resolve before restoring layout.
+   * Restore the widgets of a particular instance tracker.
+   *
+   * @param tracker - The instance tracker whose widgets will be restored.
    *
-   * #### Notes
-   * This function should only be called before the `first` promise passed in
-   * at instantiation has resolved. See the notes for `LayoutRestorer.IOptions`.
+   * @param options - The restoration options.
+   */
+  restore(tracker: InstanceTracker<any>, options: ILayoutRestorer.IRestoreOptions): void;
```

Comment:
If you make this and the interfaced generic typed,  you can avoid all of the casts.